Updated hardware-selection.md (#1511)
jellyfin/jellyfin.org: docs/general/administration/hardware-selection.md

If you are planning to use a dedicated graphics card (including upgrading an old system with a dedicated GPU), the following specs are recommended:

- CPU: Intel Core i5-2300， AMD FX-8100 or better (Geekbench 6 Multicore 1500 or better), CPU Vendor / Performance will **NOT** affect hardware encode speed or quality

> - RAM: 8GB (4GB should be enough for a Linux server without a native GUI)

Reviewer: The gui part is about something like Gnome or KDE

Author: Yes, a desktop environment, however I've seen newbies query whether browser-based administrative panels are also recommended against as "GUI" is a little vague. :- )

Reviewer 2: Maybe "without a graphical desktop" or similar?

### Low Power Servers

For users with expensive electricity, or running battery-powered servers:

- Intel 12th gen or above N series platforms
- Apple M Series Mac mini
- Rockchip RK3588 / RK3588S SBC (**Advanced Users Only**)

### Potentially Problematic Hardware

These hardware platforms might lead to a poor Jellyfin experience. Please be careful to avoid them when shopping for hardware.

- Intel "Atom" CPUs: Intel J/M/N/Y series low power CPUs up to 11th gen use a different architecture than higher end parts, leading to subpar performance despite what their names might suggest. It is advised to avoid using these parts in a Jellyfin Server.

> - Consumer-grade Prebuilt NAS Appliances: The software environment on most consumer-grade prebuilt NAS appliances often makes third-party software more challenging to install, sometimes even preventing it from working properly despite a successful installation. They may have low-end processors that are too slow for an acceptable Jellyfin experience. E.g., Intel Atom, Realtek ARM CPUs et al.

Reviewer: This actually applies to all NAS appliances. Some of the enterprise ones even though they have more powerful CPUs might still not have a GPU. 

The weird software environment also holds.

- Transcoding Audio
- Decoding video in unsupported codecs

> Audio codecs are very lightweight and most CPUs should be able to handle them without issues, e.g., Ryzen 5950X `FLAC -> AAC` transcoding runs at ~110x real-time. Please note that they will only use a single core.

Reviewer: This is not really an example, more of providing additional context on what kinds of speed we are looking at here

In BIOS settings of motherboards, it may also be called `Smart Access Memory` or `Clever Access Memory`

When using Intel ARC Graphics, Resizable BAR is recommended. Disabling it will result in a 10% reduction in transcoding performance. However, given how fast the media engine on Intel ARC is, this should not be a problem for the average deployment.

#### Software HDR to SDR Tone-mapping

If there is no GPU available on a Jellyfin Server, the CPU can also be used to tone-map HDR content to SDR. **This can be very demanding on
the CPU, therefore a GPU is ALWAYS recommended.** When using software tone-mapping, a software encoder will always be
used. Tone-mapping 4K 60fps Dolby Vision content to 4K 60fps H264 SDR requires a Ryzen 9 5950X for faster than real-time
transcoding. You may find current desktop CPUs do not offer adequate performance if you seek to provide streams in H265 or AV1.

### System Memory (RAM)

> As a general recommendation, at least 8GB is recommended for the average deployment. For a Linux server running without a GUI, 4GB may be adequate.

Reviewer: Use "may be enough" since we have users whose primary language isn't English

- Intel drivers and the compute environment is much easier to setup than both NVIDIA and AMD

AMD is not recommended due to poor quality H.264 and H.265 (HEVC) output, as well as being hard to set up the compute environment. While AMD has significantly improved AV1 encoder quality, you are still more likely to transcode to H.264 or H.265 than to AV1 due to compatibility.

A list of common codecs can be found in the [codec support documentation](/docs/general/clients/codec-support/).

The following is a list of video codecs Jellyfin supports transcoding to:

- H.264 (most common transcode target)
- H.265 (Jellyfin clients have limited support)
- AV1 (new in Jellyfin v10.9, supported by most modern browsers)

For decoding support, the more codecs the better. However, there are only a few codecs that media is commonly available in. The most important one to look for is HEVC 10bit decoding support, as it can be very demanding on the CPU to decode.

#### Intel Graphics

Please refer to [this table](https://en.wikipedia.org/wiki/Intel_Quick_Sync_Video#Hardware_decoding_and_encoding) for supported codecs.

> Intel CPUs with a model name that ends with `F` do not have integrated graphics. These are intended to be used with a dedicated graphics card. Ergo, if you wish to use Intel integrated graphics, this is not an appropriate choice of component for your Jellyfin Server.

Reviewer: Replace "Ergo" with something more understandable by non native English speakers

- Transcoding Audio
- Decoding video in unsupported codecs

> Whilst audio codecs will only utilize a single core, they are very lightweight, most CPUs should be able to handle them without issues. E.g., a Ryzen 5950X transcoding `FLAC -> AAC` runs at ~110x real-time.

Reviewer 3: ```suggestion
Whilst audio codecs will only utilize a single core, they are very lightweight and most CPUs should be able to handle them without issues. (e.g. a Ryzen 5950X transcoding `FLAC -> AAC` runs at ~110x real-time.)
```

The comma makes the sentence weird. "Whilst they use one core, they are light speed AND therefore can still be processed fast" is the wording we should be going for imo.

> Whilst audio codecs will only utilize a single core, they are very lightweight and most CPUs should be able to handle them without issues. (e.g. a Ryzen 5950X transcoding `FLAC -> AAC` runs at ~110x real-time.)

Reviewer: ```suggestion
Whilst most audio codecs will only utilize a single core, they are very lightweight and most CPUs should be able to handle them without issues. (e.g. a Ryzen 5950X transcoding `FLAC -> AAC` runs at ~110x real-time.)
```
Someone pointed out to me that some audio codecs actually have multi core implementations

Video in unsupported codecs are usually older formats that are easier to decode. Assuming hardware acceleration is properly configured, any modern CPU with 4 threads should be able to handle the workload.

However, newer codecs can also require software decoding if there are no available hardware decoders. HEVC, VP9 and AV1 can be **VERY** demanding even on modern CPUs if hardware acceleration is not available on a Jellyfin Server. This is why a hardware platform capable of HEVC 10bit hardware decoding is strongly recommended.

> In our custom `jellyfin-ffmpeg` fork, the dav1d decoder is enabled, leading to faster software decoding compared to HEVC or VP9. However, it will still use significantly more resources compared to software decoding H.264 and older codecs. Since we have a more efficient AV1 software decoder, and AV1 media isn't common yet, you shouldn't worry about AV1 when choosing components for your Jellyfin Server.

Reviewer: I actually think this part can be removed entirely, since it is a bit of a side tangent really

Author: Feel free to suggest the change so I can `commit` your want directly.

#### Integrated Graphics

Integrated graphics can be useful for transcoding video. For more information, please refer to [the GPU section](/docs/general/administration/hardware-selection#graphics-cards-gpus).

#### Resizable BAR

[Resizable Bar (ReBAR)](https://www.intel.com/content/www/us/en/support/articles/000090831/graphics.html) is a PCI Express feature that optimizes access to PCIe devices. This feature requires support from both the platform and the device. Support for this feature is respectively available beginning with Intel 10th gen and AMD Ryzen 3000 Series.

In BIOS settings of motherboards, it may also be called `Smart Access Memory` or `Clever Access Memory`

When using Intel ARC Graphics, Resizable BAR is recommended. Disabling it will result in a 10% reduction in transcoding performance. However, given how fast the media engine on Intel ARC is, this should not be a problem for the average deployment.

> 

Reviewer: Add a note here about ARC B series, the media engine will not function at all without ReBar

Author: Feel free to commit the suggestion.

With modern systems, the media engines are usually located on the GPUs. Therefore, the available hardware acceleration options are determined by the choice of GPU.

Encoder Quality: Apple ≥ Intel ≥ NVIDIA >>> AMD<sup>\*</sup>

<sup>\*</sup> This only represents the default Jellyfin settings. The quality may be different depending on your exact configuration.

Intel is always recommended on non-Apple hardware for the following reasons:

- Intel provides a good quality encoder, slightly better than NVIDIA and significantly better than AMD.
- Intel drivers and the compute environment is much easier to setup than both NVIDIA and AMD

AMD is not recommended due to poor quality H.264 and H.265 (HEVC) output, as well as being hard to set up the compute environment. While AMD has significantly improved AV1 encoder quality, you are still more likely to transcode to H.264 or H.265 than to AV1 due to the hardware capabilities of the average Jellyfin client.

A list of common codecs can be found in the [codec support documentation](/docs/general/clients/codec-support/).

The following is a list of video codecs Jellyfin supports transcoding to:

- H.264 (most common transcode target)

> - H.265 (Jellyfin clients have limited support)

Reviewer 3: The limited support does not necessarily need to be on the JELLYFIN client side. Hardware wise, client devices often do not support h265. Not an issue with the Jellyfin client applications but simply client devices in general.
Therefore I dislike the new wording here!

Author: Approved all your suggestions, though am admittedly offended by saying something I wrote seems "AI generated" without elaboration and will thus be my final contributions to the Jellyfin project.

If you intend to use Linux with Intel 12/13th gen integrated graphics or ARC, these GPUs only work on Linux Kernel 6.2 or newer. Please check your distribution to make sure it has a supported Linux Kernel version. Further information: [Known Intel limitations on Linux](/docs/general/post-install/transcoding/hardware-acceleration/known-issues#intel-on-linux).

:::caution

Intel 10th gen and older integrated graphics are losing support for QSV on Linux due to the SDK for these platforms being deprecated by Intel. If you own 7-10th gen CPUs with integrated graphics, please continue to use them for Jellyfin. If you are making a purchase decision, you may wish to consider a newer CPU if you plan on using Intel integrated graphics.

:::

#### NVIDIA Graphics

Please refer to [this table](https://developer.nvidia.com/video-encode-and-decode-gpu-support-matrix-new) for supported codecs.

Certain low-end cards, e.g., GT 1030, do not have encoding hardware. Please be careful when choosing a GPU.

#### AMD Graphics

> AMD graphics is not recommended for Jellyfin, but the information is still provided for reference. Information on supported codecs is on [this article](https://en.wikipedia.org/wiki/Video_Coding_Engine).

Reviewer: If you are going to change this for better readability you might as well change the "this article" link text to something descriptive

Author: Updated it: https://github.com/jellyfin/jellyfin.org/pull/1511/commits/b457979ee03abcb920c7aea89f1218c429e9ebf6

✅ = Encode + Decode, 🔶 = Decode Only, ❌ = Not Supported.

<sup>1</sup> Although the hardware does support AV1 decoding, [ffmpeg does not yet support it](https://trac.ffmpeg.org/ticket/10642).

:::caution

> No Apple Silicon media engine drivers currently exist for non-macOS operating systems. You will NOT be able to use hardware acceleration if you are running [Asahi Linux](https://asahilinux.org/) on your Apple Silicon hardware.

Reviewer: "On your apple silicon hardware" is redundant because it can't run on anything else

Author: Rewritten it for clarity: https://github.com/jellyfin/jellyfin.org/pull/1511/commits/5cdc9679116c6a9cc4103e0a0e0886d4085da158

### Storage

> There are two general types of storage that can be used on a Jellyfin Server. Media files and Jellyfin files each require different performance characteristics for optimal performance.

Reviewer: The "each" is redundant

Author: Rewritten it for clarity: https://github.com/jellyfin/jellyfin.org/pull/1511/commits/e5221603a59c4cb414377ac22a8e60717d52d4b1

#### Media Storage

> Anything with higher sequential access speed than bitrate of media is appropriate. If desired, this storage may be a network share, or a cloud storage service mounted on the Jellyfin Server host.

Reviewer: "Is acceptable" is closer to the meaning than "is appropriate".

Author: "adequate" is likely a better word than either.

Author: With that said, the page title is `Selecting Appropriate Hardware`, and the first paragraph states the goal as `This page covers what you need to know in order to select appropriate hardware for a Jellyfin server and take full advantage of its features (e. g. hardware acceleration).`.

Reviewer 3: OG wording totaly fine imo. Marking as resolved.

#### Jellyfin Files

The files for Jellyfin itself (i.e., not your media files) will see a lot of random access, therefore SSDs are recommended for these files. AVOID mechanical drives that use [SMR (Shingled Magnetic Recording)](https://en.wikipedia.org/wiki/Shingled_magnetic_recording) as they will result in a VERY poor experience.

### Networking

Networking is for connecting your Jellyfin server to other devices. It is recommended that the Jellyfin Server be connected to your network via Ethernet cable for optimal performance. Wi-Fi or Powerline solutions for a Jellyfin Server are NOT recommended and will result in an especially poor experience.

> Jellyfin Server is not designed to be exposed directly to the internet, do NOT accidentally connect your server to ethernet ports marked `DMZ` on your network router. Always check the port label before inserting the cable.

Reviewer: The DMZ configuration is usually in software now. If someone has a device that is physically marked DMZ then it is pro grade gear and they will know what they are doing.

Reviewer 3: Maybe have it more generally phrased.
This way the point still is made, but the physical DMZ (which I personaly also have never seen) is not explicitly mentioned.

```suggestion
Jellyfin Server is not designed to be exposed directly to the internet. Ensure it remains protected behind a properly configured firewall or other secure network environment.
```

What do you think?
